CONTRIBUTING.adoc

```diff
@@ -108,55 +108,78 @@ NOTE: you must run your editor from the nix shell, and not from, say, an
+packages. This means that downstream consumers have no idea of your
+`index-state` setting and you need to make sure your packages work
+correctly also without `index-state`.
 
-[[update-haskell-pins]]
-=== How to update our pinned Haskell dependencies
+Because of how we use Nix to manage our Haskell build, whenever you do this
+you will also need to pull in the Nix equivalent of the newer
+``index-state``. You can do this by running ``nix flake lock --update-input
+hackage-nix``.
 
-We have pinned versions of some Haskell packages specified via the usual `source-repository-package` (Cabal) and `extra-dep` (Stack) mechanisms.
+==== ... from the Cardano package repository
 
-These can be managed normally, but ensure that:
+Many Cardano packages are not on Hackage and are instead in the `Cardano
+package repository
+<https://github.com/input-output-hk/cardano-haskell-packages>`, see the
+README for (lots) more information.
 
-* If it is an `source-repository-package`/`extra-dep` from Git, you update the `sha256` mapping in `nix/haskell.nix`.
-For the moment you have to do this by hand, using the following command to get the sha: `nix-prefetch-git --quiet <repo-url> <rev> | jq .sha256`, or by just getting it wrong and trying to build it, in which case Nix will give you the right value.
-* For certain dependencies like `plutus-apps`, please run `nix flake lock --update-input <input-name>` to update `flake.lock`. A wrong-hash error on a `nix build` is a common hint.
+Getting new packages from there works much like getting them from Hackage.
+The differences are that it can have an independent ``index-state``, and
+that there is a different Nix command you need to run afterwards: ``nix flake
+lock --update-input CHaP``.
 
-[[update-nix-pins]]
-=== How to update our pinned Nix dependencies
+===== Using unreleased versions of dependencies
 
-We pin versions of some git repositories that are used by Nix, for example `nixpkgs`.
+Sometimes we need to use an unreleased version of one of our dependencies,
+either to fix an issue in a package that is not under our control, or to
+experiment with a pre-release version of one of our own packages.
 
-We use Nix flakes to manage these dependencies, even though we do not
-actually use flakes for normal Nix builds. To manage the dependencies,
-see https://nixos.org/manual/nix/unstable/command-ref/new-cli/nix3-flake.html#flake-inputs[the Nix flake inputs documentation]
-and https://nixos.org/manual/nix/unstable/command-ref/new-cli/nix3-flake-lock.html[the Nix flake lock command].
+You can use a ``source-repository-package`` stanza to pull in the
+unreleased version.
 
-Specifically, you will probabl want to say `nix flake lock --update-input <input-name>`.
-Do *not* use `nix flake update`, as that will update all the inputs, which we typically don't want to do.
+Try only to do this for a short time, as it does not play very well with
+tooling, and will interfere with the ability to release the node itself.
```

Comment:
I don't think that's relevant for plutus-apps

Reply:
Sorry bad copy & paste. Fixed